evm/src/executor/inspector/coverage.rs

```diff
@@ -0,0 +1,139 @@
+use crate::{coverage::HitMaps, executor::inspector::utils::get_create_address};
+use bytes::Bytes;
+use ethers::types::Address;
+use revm::{
+    opcode, spec_opcode_gas, CallInputs, CreateInputs, Database, EVMData, Gas, Inspector,
+    Interpreter, Return, SpecId,
+};
+use std::collections::BTreeMap;
+
+#[derive(Default, Debug)]
+pub struct CoverageCollector {
+    /// Maps that track instruction hit data.
+    pub maps: HitMaps,
+
+    /// The execution addresses, with the topmost one being the current address.
+    context: Vec<Address>,
+
+    /// A mapping of program counters to instruction counters.
+    ///
+    /// The program counter keeps track of where we are in the contract bytecode as a whole,
+    /// including push bytes, while the instruction counter ignores push bytes.
+    ///
+    /// The instruction counter is used in Solidity source maps.
+    pub ic_map: BTreeMap<Address, BTreeMap<usize, usize>>,
```

Comment:
does this need to be sorted?

Reply:
Not really, but the alternative is that we hash the small integers which I think is a bigger performance penalty?

Reply:
simple hashmap should also do it

Reply:
Yeah but that's what I mean - the HashMap will hash the usizes and this can be pretty slow considering how small the usizes really are. I don't think BTreeMap hashes?